src/main/java/com/xpn/xwiki/plugin/lucene/IndexUpdater.java

```diff
@@ -245,7 +245,7 @@ private void updateIndex() {
         getContext().getWiki().getStore().cleanUp(getContext());
         try {
           writer.optimize();
-          writer.close();
+          IOUtils.closeQuietly(writer);
```

Comment:
closeQuietly must be outside try catch block. Because if optimize fails with an IOException, the writer will not be closed correctly.